### PR TITLE
chore: update doc for `partial_chain` and `verify_peer_ext_key_usage`

### PR DESCRIPTION
**Pull request by qzhuyan**
followup review comments in https://github.com/emqx/emqx/pull/12955

Release version: v/e5.8
## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: changes/ce/feat-11721.en.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
-Port two TLS handshake validation features from emqx 4.4
+Enhance TLS listener to support more flexible TLS verifications.
 
 - partial_chain support
+
+  If the option `partial_chain` is set to `true`, allow connections with incomplete certificate chains.
+  
+  Check the description in emqx schema for more.
```
Comment (zmstone) [inline changes/ce/feat-11721.en.md:7]: Schema presented to user as "configuration manual document".
in this format: https://docs.emqx.com/en/enterprise/v5.6.1/hocon/
```suggestion
  Check the configuration manual docment for more details.
```

File: changes/ce/feat-11721.en.md [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+  [rfc5280](https://www.rfc-editor.org/rfc/rfc5280#section-4.2.1.12).
+
+  Introduced a new option (`verify_peer_ext_key_usage`) to require specific key usages (like "serverAuth") 
+  in peer certificates during the TLS handshake.
+  This strengthens security by ensuring certificates are used for their intended purposes.
+
+  example:
+     "serverAuth,OID:1.3.6.1.5.5.7.3.2"
+    
+  Check the description in emqx schema for more.
```
Comment (zmstone) [inline changes/ce/feat-11721.en.md:21]:
```suggestion
  Check the configuration manual document for more details.
```

File: rel/i18n/emqx_schema.hocon [hunk 3]
```diff
@@ -679,13 +679,45 @@ common_ssl_opts_schema_verify.label:
...
+However, if the setting is set to `true` or `cacert_from_cacertfile`,
+the last certificate in the cacertfile will be used as the trust anchor
+certificate (such as an intermediate CA). This creates a partial chain
+in the path validation.
+Alternatively, if the setting is set to `two_cacerts_from_cacertfile`,
+one of the last two certificates in the cacertfile will be used as the
+trust anchor certificate, forming a partial chain. This option is
+particularly useful for CA certificate rotation.
+However, please note that it incurs some additional overhead, so it
+should only be used for certificate rotation purposes."""
```
Comment (zmstone) [inline rel/i18n/emqx_schema.hocon:701]: if the description cannot be split to bullets (one for each enum symbol), then there is no need to repeat the enums as is.

also some refine suggestions
```suggestion
"""Enable or disable peer verification with partial_chain.

When local verifies a peer certificate during the x509 path validation
process, it constructs a certificate chain that starts with the peer
certificate and ends with a trust anchor.
By default, if it is set to `false`, the trust anchor is the
Root CA, and the certificate chain must be complete.
However, if the setting is set to `true` or `cacert_from_cacertfile`,
the last certificate in `cacertfile` will be used as the trust anchor
certificate (intermediate CA). This creates a partial chain
in the path validation.
Alternatively, if it is configured with `two_cacerts_from_cacertfile`,
one of the last two certificates in `cacertfile` will be used as the
trust anchor certificate, forming a partial chain. This option is
particularly useful for intermediate CA certificate rotation.
However, please note that it incurs some additional overhead, so it
should only be used for certificate rotation purposes."""
```

File: rel/i18n/emqx_schema.hocon [hunk 4]
```diff
@@ -679,13 +679,45 @@ common_ssl_opts_schema_verify.label:
...
 
 common_ssl_opts_schema_partial_chain.label:
 """Partial chain"""
 
 common_ssl_opts_verify_peer_ext_key_usage.desc:
-"""Verify Extended Key Usage in Peer's certificate"""
+"""Verify Extended Key Usage in Peer's certificate
```
Comment (zmstone) [inline rel/i18n/emqx_schema.hocon:707]:
```suggestion
"""Verify extended key usage in peer's certificate
```

File: rel/i18n/emqx_schema.hocon [hunk 5]
```diff
@@ -679,13 +679,45 @@ common_ssl_opts_schema_verify.label:
...
+[rfc5280](https://www.rfc-editor.org/rfc/rfc5280#section-4.2.1.12).
+Allowed values are
+- "clientAuth"
+- "serverAuth"
+- "codeSigning"
+- "emailProtection"
+- "timeStamping"
+- "ocspSigning"
+- raw OID, for example: "OID:1.3.6.1.5.5.7.3.2"
+Comma-separated string is also supported for validating the subset of key usages.
```
Comment (zmstone) [inline rel/i18n/emqx_schema.hocon:719]:
```suggestion
Comma-separated string is also supported for validating more than one key usages.
```

File: rel/i18n/emqx_schema.hocon [hunk 6]
```diff
@@ -679,13 +679,45 @@ common_ssl_opts_schema_verify.label:
 """Verify peer"""
 
 common_ssl_opts_schema_partial_chain.desc:
-"""Enable or disable peer verification with partial_chain"""
+"""Enable or disable peer verification with partial_chain:
+- `false`
+- `true`
+- `cacert_from_cacertfile`
+- `two_cacerts_from_cacertfile`
```
Comment (qzhuyan) [inline rel/i18n/emqx_schema.hocon:686]: need line break

File: rel/i18n/emqx_schema.hocon [hunk 7]
```diff
@@ -679,13 +679,45 @@ common_ssl_opts_schema_verify.label:
...
+'Extended Key Usage' of peer certificate defined in
+[rfc5280](https://www.rfc-editor.org/rfc/rfc5280#section-4.2.1.12).
+Allowed values are
+- "clientAuth"
+- "serverAuth"
+- "codeSigning"
+- "emailProtection"
+- "timeStamping"
+- "ocspSigning"
+- raw OID, for example: "OID:1.3.6.1.5.5.7.3.2"
```
Comment (zmstone) [inline rel/i18n/emqx_schema.hocon:718]:
```suggestion
- raw OID, for example: "OID:1.3.6.1.5.5.7.3.2" means `id-pk 2` which is equivalent to `clientAuth`
```

File: rel/i18n/emqx_schema.hocon [hunk 8]
```diff
@@ -679,13 +679,52 @@ common_ssl_opts_schema_verify.label:
 """Verify peer"""
 
 common_ssl_opts_schema_partial_chain.desc:
-"""Enable or disable peer verification with partial_chain"""
+"""Enable or disable peer verification with partial_chain:
```
Comment (qzhuyan) [inline rel/i18n/emqx_schema.hocon:682]:
```suggestion
"""Enable or disable peer verification with partial_chain.
When local verifies a peer certificate during the x509 path validation
process, it constructs a certificate chain that starts with the peer
certificate and ends with a trust anchor.
By default, if it is set to `false`, the trust anchor is the
Root CA, and the certificate chain must be complete.
However, if the setting is set to `true` or `cacert_from_cacertfile`,
the last certificate in `cacertfile` will be used as the trust anchor
certificate (intermediate CA). This creates a partial chain
in the path validation.
Alternatively, if it is configured with `two_cacerts_from_cacertfile`,
one of the last two certificates in `cacertfile` will be used as the
trust anchor certificate, forming a partial chain. This option is
particularly useful for intermediate CA certificate rotation.
However, please note that it incurs some additional overhead, so it
should only be used for certificate rotation purposes."""
```

**Review by zmstone — DISMISSED** [review]

File: scripts/spellcheck/dicts/emqx.txt [hunk 9]
```diff
@@ -307,3 +307,10 @@ ElasticSearch
 doc_as_upsert
 upsert
 aliyun
+rootCA
+clientAuth
+serverAuth
+codeSigning
+emailProtection
+ocspSigning
```
Comment (zmstone) [inline scripts/spellcheck/dicts/emqx.txt:315]: nit: no need to add them if the usage is quoted between a pair of `

File: scripts/spellcheck/dicts/emqx.txt [hunk 10]
```diff
@@ -307,3 +307,10 @@ ElasticSearch
 doc_as_upsert
 upsert
 aliyun
+rootCA
```
Comment (zmstone) [inline scripts/spellcheck/dicts/emqx.txt:310]: nit: rootCA is no longer used after the new commit.

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2507)

**Pull request merged**